components/data/data.c

```diff
@@ -1,17 +1,19 @@
 #include <stdio.h>
 #include "data.h"
 
-csha_bt_packet* new_frame()
-{
-    return malloc(sizeof(csha_bt_packet));
-}
 
 int calc_len(csha_bt_packet* frame)
 {
-    return strlen(frame->name) + strlen(frame->mac) + (frame->rssi == 0 || frame->rssi == 1 ? 1 : ceil(log10(frame->rssi))) + 5 + 17;
+    return strlen(frame->name) + strlen(frame->mac) + (frame->rssi == 0 || frame->rssi == 1 ? 1 : ceil(log10(abs(frame->rssi)))) + (frame->rssi < 0 ? 1:0) + 5 + 18 + 18 + 10;
```

Comment:
Yeah this bit is pretty scuffed.  It's just meant to give the shortest character amount required to hold all of the data in a formatted UDP string.  It's actually not 100% correct at the moment, it overshoots a bit ever since I added timestamps.

Reply:
Haha cool